### PR TITLE
chore(infra): run tests on main to get code coverage

### PR DESCRIPTION
**Pull request by dabbeg**
## What

Run tests on main

## Why

So we get the test coverage to upload to CodeCov

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: scripts/download-codecov-uploader.sh [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #!/usr/bin/env bash
```
Comment (sindrig) [inline scripts/download-codecov-uploader.sh:1]: how long does this usually take? is it worth it to use the cache considering the time difference and the added boilerplate?

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -277,6 +305,58 @@ jobs:
           UNAFFECTED=$(./scripts/ci/list-unaffected.sh docker-next docker-express docker-static docker-cypress)
           echo "::set-output name=UNAFFECTED::$UNAFFECTED"
 
+  tests:
```
Comment (sindrig) [inline .github/workflows/push.yml:308]: Consider moving this to a [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action)

Comment (pshomov) [reply .github/workflows/push.yml:308]: @sindrig I find it disappointing that composite actions do not support conditional steps. I guess you can use scripting when having a shell command but kind of hard when you are `use`-ing an action.

File: scripts/download-codecov-uploader.sh (hunk 1 above)
Comment (pshomov) [reply scripts/download-codecov-uploader.sh:1]: 61 MB binary

**Review by pshomov — APPROVED** [review]
LGTM, the cache for the codecov binary might be a bit too much, cache can be invalidated between jobs (but not after our 🦄  S3 cache)

File: scripts/download-codecov-uploader.sh (hunk 1 above)
Comment (sindrig) [reply scripts/download-codecov-uploader.sh:1]: caching it in azure vs downloading it from some cdn /shrug

Comment (dabbeg) [reply scripts/download-codecov-uploader.sh:1]: It took 4s when I tested this in the pipeline. I was fooled by my very slow internet connection.
I am removing the caching

Comment (sindrig) [reply scripts/download-codecov-uploader.sh:1]: image

Comment (sindrig) [reply scripts/download-codecov-uploader.sh:1]: denmark

**Pull request merged**